Core/test/qa-functional/src/org/sleuthkit/autopsy/ingest/EmbeddedFileTest.java

```diff
@@ -0,0 +1,194 @@
+        } catch (CaseActionException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        }
+        assertTrue(CASE_DIR.exists());
+        ImageDSProcessor dataSourceProcessor = new ImageDSProcessor();
+        try {
+            DataSourceProcessorRunner.ProcessorCallback callBack = DataSourceProcessorRunner.runDataSourceProcessor(dataSourceProcessor, IMAGE_PATH);
+            List<Content> dataSourceContent = callBack.getDataSourceContent();
+            assertEquals(1, dataSourceContent.size());
+            List<String> errorMessages = callBack.getErrorMessages();
+            assertEquals(0, errorMessages.size());
+        } catch (AutoIngestDataSourceProcessor.AutoIngestDataSourceProcessorException | InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+            
+        }
+        
+        try {
+            openCase = Case.getOpenCase();
+            runIngestJob(openCase.getDataSources());
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        } 
+    }
+
+    @Override
+    public void tearDown() {
+        try {
+            Case.closeCurrentCase();
+            //Seems like we need some time to close the case.
+            try {
+                Thread.sleep(2000);
+            } catch (Exception ex) {
+
+            }
+
+            FileUtils.deleteDirectory(CASE_DIR);
+        } catch (CaseActionException | IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        assertFalse(CASE_DIR.exists());
+    }
+    
+    public void testEncription() {
+        try {
+            List<AbstractFile> results = openCase.getSleuthkitCase().findAllFilesWhere("name LIKE '%%'");            
+            String protectedName1 = "password_protected.zip";
+            String protectedName2 = "level1_protected.zip";
+            String protectedName3 =  "42.zip";
+            assertEquals(2207, results.size());
+            int passwdProtectedZips = 0;
+            for (AbstractFile file : results) {
+                //.zip file has artifact TSK_ENCRYPTION_DETECTED
+                if (file.getName().equalsIgnoreCase(protectedName1) || file.getName().equalsIgnoreCase(protectedName2) || file.getName().equalsIgnoreCase(protectedName3)){
+                    ArrayList<BlackboardArtifact> artifacts = file.getAllArtifacts();
+                    assertEquals(1, artifacts.size());
+                    for (BlackboardArtifact artifact : artifacts) {
+                        assertEquals(artifact.getArtifactTypeID(), BlackboardArtifact.ARTIFACT_TYPE.TSK_ENCRYPTION_DETECTED.getTypeID());
+                        passwdProtectedZips++;
+                    }
+                } else {//No other files has artifact defined
```

Comment:
Please change "No other files has artifact defined" to "No other files have artifacts."